Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/WindowsKeyboard/MixedRealityKeyboard.cs

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.Events;
+
+#if !UNITY_EDITOR && UNITY_WSA
+using Windows.UI.ViewManagement;
+#endif 
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// Class that can launch and hide a system keyboard specifically for Windows Mixed Reality
+    /// devices (HoloLens 2, Windows Mixed Reality).
+    /// 
+    /// Implements a workaround for UWP TouchScreenKeyboard bug which prevents
+    /// UWP keyboard from showing up again after it is closed.
+    /// Unity bug tracking the issue https://fogbugz.unity3d.com/default.asp?1137074_rttdnt8t1lccmtd3
+    /// </summary>
+    public class MixedRealityKeyboard : MonoBehaviour
+    {
+        [Experimental]
+        #region Serialized fields
+        [SerializeField]
+        [Tooltip("Event raised when the text has changed")]
+        public KeyboardTextEvent TextChanged = new KeyboardTextEvent();
+        [Tooltip("Event raised when the keyboard is shown")]
+        public UnityEvent KeyboardShown = new UnityEvent();
+        [Tooltip("Event raised when the keyboard is hidden")]
+        public UnityEvent KeyboardHidden = new UnityEvent();
+        #endregion
+
+        #region Properties
+        private string text;
+        public string Text
+        {
+            get
+            {
+                return text;
+            }
+
+            private set
+            {
+                if (text != value)
+                {
+                    var oldValue = text;
+                    text = value;
+                    TextChanged?.Invoke(new KeyboardTextData(oldValue, value));
```

Comment:
>TextChanged [](start = 20, length = 11)

This aspect of this file is more than just a show/hide wrapper for the TouchScreenKeyboard - i.e. at this point, it's no longer just a workaround for a buggy underlying API - this actually adds new features, which means that even once Unity fixes the underlying buggy API technically speaking, this will not be deletable at that point.

If there is significant ask/value to this, then this is fine, just note that this is sort of a "hidden" feature that is sneaking in with this change.

Reply:
I guess technically since it's in experimental we could delete it then, I just wanted to call this out because it isn't mentioned in the PR description.

---
In reply to: [296405679](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5018#discussion_r296405679) [](ancestors = 296405679)

Reply:
Or just update the PR description + comments on the code to mention this as well, either/or is fine by me.

Reply:
I removed all the handy events so it's a simple class. Makes sense since we wouldn't want to break people once we removed it.